### PR TITLE
feat: integrate formbricks for team disband action

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
- Renames existing Formbricks env variables to work client side too
- Adds a new `formbricks/js` package
- Builds a new `formbricks-client.ts` file to run client side `formbricks/js` methods
- Tracks the `team_disbanded` whenever the deleteTeam happens

## Type of change
- New feature (non-breaking change which adds functionality)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/ee/organizations/pages/components/OtherTeamListItem.tsx [hunk 1]
```diff
@@ -143,6 +144,7 @@ export default function OtherTeamListItem(props: Props) {
                         isLoading={props.isLoading}
                         onConfirm={() => {
                           props.onActionSelect("disband");
+                          trackFormbricksAction("team_disbanded");
```
Comment (mattinannt) [inline packages/features/ee/organizations/pages/components/OtherTeamListItem.tsx]: is this the same code part as the file above; may this event be triggered twice?

File: packages/features/ee/teams/components/TeamListItem.tsx [hunk 2]
```diff
@@ -325,6 +326,7 @@ export default function TeamListItem(props: Props) {
                             isLoading={props.isLoading}
                             onConfirm={() => {
                               props.onActionSelect("disband");
+                              trackFormbricksAction("team_disbanded");
```
Comment (mattinannt) [inline packages/features/ee/teams/components/TeamListItem.tsx]: same question here.

File: packages/lib/formbricks-client.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { useEffect } from "react";
+
+import useMeQuery from "@calcom/trpc/react/hooks/useMeQuery";
+
+export const initFormbricks = ({ userId, attributes }: { userId: string; attributes: any }) => {
+  if (process.env.NEXT_PUBLIC_FORMBRICKS_HOST_URL && process.env.NEXT_PUBLIC_FORMBRICKS_ENVIRONMENT_ID) {
+    formbricks.init({
+      environmentId: process.env.NEXT_PUBLIC_FORMBRICKS_ENVIRONMENT_ID,
+      apiHost: process.env.NEXT_PUBLIC_FORMBRICKS_HOST_URL,
+      debug: true,
```
Comment (mattinannt) [inline packages/lib/formbricks-client.ts]: Can we enable this only in development; e.g. `debug: process.env.NODE_ENV === 'development'`
(if this is working client side)

File: packages/lib/formbricks-client.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import formbricks from "@formbricks/js";
+import { useSession } from "next-auth/react";
+import { useEffect } from "react";
+
+import useMeQuery from "@calcom/trpc/react/hooks/useMeQuery";
+
+export const initFormbricks = ({ userId, attributes }: { userId: string; attributes: any }) => {
+  if (process.env.NEXT_PUBLIC_FORMBRICKS_HOST_URL && process.env.NEXT_PUBLIC_FORMBRICKS_ENVIRONMENT_ID) {
```
Comment (mattinannt) [inline packages/lib/formbricks-client.ts:21]: Can we here maybe make a variable
```
const isFormbricksEnabled = process.env.NEXT_PUBLIC_FORMBRICKS_HOST_URL && process.env.NEXT_PUBLIC_FORMBRICKS_ENVIRONMENT_ID
```

**Review by mattinannt — COMMENTED** [review]
@ShubhamPalriwala Thanks you :-) Looks great! 😊
A few questions I have when looking at the code:

File: packages/lib/formbricks-client.ts (hunk 4 above)
Comment (ShubhamPalriwala) [reply packages/lib/formbricks-client.ts:21]: Hey, tried doing so as:
```ts
const isFormbricksEnabled =
  process.env.NEXT_PUBLIC_FORMBRICKS_HOST_URL && process.env.NEXT_PUBLIC_FORMBRICKS_ENVIRONMENT_ID;

export const initFormbricks = ({
  userId,
  attributes,
}: {
  userId: string;
  attributes: { [key: string]: string };
}) => {
  if (isFormbricksEnabled) {
    formbricks.init({
      environmentId: process.env.NEXT_PUBLIC_FORMBRICKS_ENVIRONMENT_ID,
      apiHost: process.env.NEXT_PUBLIC_FORMBRICKS_HOST_URL,
      debug: process.env.NODE_ENV === "development",
      userId,
      attributes,
    });
  }
};
```
but ran into a build error:
```
 Type error: Type 'string | undefined' is not assignable to type 'string'.
@calcom/web:build:   Type 'undefined' is not assignable to type 'string'.
@calcom/web:build: 
@calcom/web:build:   17 |   if (isFormbricksEnabled) {
@calcom/web:build:   18 |     formbricks.init({
@calcom/web:build: > 19 |       environmentId: process.env.NEXT_PUBLIC_FORMBRICKS_ENVIRONMENT_ID,
@calcom/web:build:      |       ^
@calcom/web:build:   20 |       apiHost: process.env.NEXT_PUBLIC_FORMBRICKS_HOST_URL,
@calcom/web:build:   21 |       debug: process.env.NODE_ENV === "development",
@calcom/web:build:   22 |       userId,
```

GPT said:
You're right, you have a check in place with isFormbricksEnabled to ensure that process.env.NEXT_PUBLIC_FORMBRICKS_HOST_URL and process.env.NEXT_PUBLIC_FORMBRICKS_ENVIRONMENT_ID are defined before using them in the formbricks.init function. However, TypeScript's static type checker doesn't infer that these checks guarantee the non-nullability of the environment variables within the formbricks.init call.

TypeScript's type system is based on the types of variables and expressions at compile time, and it doesn't execute or evaluate the logic of your code. So even though you have a runtime check with isFormbricksEnabled, TypeScript still sees process.env.NEXT_PUBLIC_FORMBRICKS_ENVIRONMENT_ID and process.env.NEXT_PUBLIC_FORMBRICKS_HOST_URL as potentially undefined.

And using the non-null assertion (!) is not allowed in their tsconfig so looks like we might need to keep things like this.

Comment (mattinannt) [reply packages/lib/formbricks-client.ts:21]: Ok, makes sense. Then let's stick to the current approach :-)

**Pull request closed without merging**